Prometheus auto service discovery

maulindesai opened the pull request:
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [X] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [X] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [X] Images use HTML style and have alternative text
- [X] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

- Updated prometheus.md with instructions for setting up Prometheus auto service discovery for Dapr and sidecar targets.

## Issue reference
#4234
<!--Please reference the issue this PR will close: #[issue number]-->

Commit:
Add Prometheus auto service discovery instructions to
prometheus.md

- Updated prometheus.md with instructions for setting up Prometheus auto service discovery for Dapr and sidecar targets.

Commit:
Add Prometheus auto service discovery instructions to
prometheus.md

- Updated prometheus.md with instructions for setting up Prometheus auto service discovery for Dapr and sidecar targets.

Signed-off-by: Maulin Desai <mdesai@bosleo.com>

Commit:
Merge remote-tracking branch 'origin/prometheus-auto-service-discovery' into prometheus-auto-service-discovery

hhunter-ms commented on daprdocs/content/en/operations/observability/metrics/prometheus.md:
````diff
@@ -110,6 +205,22 @@ dapr-prom-prometheus-pushgateway-688665d597-h4xx2   1/1     Running   0
+kubectl port-forward svc/dapr-prom-prometheus-server 9090:80 -n dapr-monitoring
+```
+
+Open a browser and visit `http://localhost:9090`. Navigate to `Status -> Service Discovery` to verify that the Dapr targets are discovered correctly.
````
```suggestion
Open a browser and visit `http://localhost:9090`. Navigate to **Status** > **Service Discovery** to verify that the Dapr targets are discovered correctly.
```

hhunter-ms reviewed (changes requested):
quick formatting fix to keep consistent w other docs. otherwise lgtm!

Commit:
Format service discovery instructions

Updated the navigation instructions to use bold text for "Status" and "Service Discovery" for better visual clarity.

Signed-off-by: Maulin Desai <mdesai@bosleo.com>

Commit:
Merge branch 'v1.14' into prometheus-auto-service-discovery

hhunter-ms committed:
Merge branch 'v1.14' into prometheus-auto-service-discovery

msfussell reviewed (approved):
LGTM

msfussell committed:
Merge branch 'v1.14' into prometheus-auto-service-discovery

marcduiker commented:
@holopin-bot @maulindesai Thank you for your contribution!
